gnd/src/main/java/com/google/android/gnd/persistence/local/room/RoomLocalDataStore.java

```diff
@@ -583,4 +604,50 @@ public Completable insertOrUpdateOfflineArea(OfflineArea area) {
         .toSingle()
         .subscribeOn(schedulers.io());
   }
+
+  private Completable deleteTiles(OfflineAreaEntity offlineAreaEntity) {
+    return tileSourceDao
+        .getTileSourcesWithOfflineAreas()
+        .toSingle()
+        .map(
+            sources ->
+                stream(sources)
+                    .filter(
+                        source ->
+                            // We only remove tiles when the given area is the only one that depends
+                            // on them.
+                            source.offlineAreaEntities.contains(offlineAreaEntity)
+                                && source.offlineAreaEntities.size() == 1)
+                    .map(source -> source.tileSourceEntity)
+                    .collect(toImmutableList()))
+        .flatMapObservable(Observable::fromIterable)
```

Comment:
Iirc there are some iterable Rx operations, like `flatMapIterable`. Do any of those help here?